docs/modules/ROOT/pages/explanations/decisions/capacity-alerting.adoc

```diff
@@ -0,0 +1,111 @@
+apiVersion: vshn.appcat.vshn.io/v1
+kind: VSHNPostgreSQL
+metadata:
+  name: pgsql-app1-prod
+  namespace: prod-app
+spec:
+  parameters:
+    alerting:
+      receivers:
+      - name: slack
+        slackConfigs:
+        - apiURL:
+            key: url
+            name: slack-alert-webhook-secret
+          channel: '#alerts'
+      route:
+        groupBy: [alertname]
+        receiver: slack
+  writeConnectionSecretToRef:
+    name: postgres-creds
+----
+
+=== Send E-Mail Alert through Alert-Manager and mailgun
+
+VSHN has the option to send e-mails automatically through https://www.mailgun.com/[mailgun].
+This can be leveraged to inform service users through e-mail without a lot of setup overhead.
+
+In practice this can be implemented fairly similarly to forwarding alerts to Opsgenie.
+The composition will create an `AlertManagertConfig` resource in the database namespace that configures alert-manager to send mails to the provided e-mail address.
+
+With such a system the database claim could look similar to the following, which would send all alerts as mails to `oncall@example.com`
+[source,yaml]
+----
+apiVersion: vshn.appcat.vshn.io/v1
+kind: VSHNPostgreSQL
+metadata:
+  name: pgsql-app1-prod
+  namespace: prod-app
+spec:
+  parameters:
+    alerting:
+      email: oncall@example.com
+  writeConnectionSecretToRef:
+    name: postgres-creds
+----
+
+There are a few things we'll need to consider for such an implementation:
+
+* This setup forces us to replicate a mailgun token to every database namespace which increases the potential attack surface.
+* We might need to think about verifying the e-mail before forwarding alerts to it.
+This could be done through a custom controller.
+
+== Decision
+
+We implement both exposing the user-workload monitoring and automated e-mail alerts through mailgun.
+E-mail based alerting should be strong encouraged, and might be mandatory in some cases, to ensure that we can prove that the service user was informed of the issue.
+
+=== Rationale
+
+Manual intervention for capacity alerts generates a lot of unnecessary toil, as in most cases the engineer can't do anything but inform the service user.
+That's why we believe that the task should be automated, and no engineer should be alerted for potential capacity issues.
+
+Exposing the whole capabilities of the user-workload monitoring system is fairly simple to implement, while at the same time very versatile.
+It allows our service users to send alerts exactly where they are needed.
+
+On the other hand automatic e-mail alerts through mailgun has a much better UX for simple deployments.
+With this solution, service users don't need to handle any mail gateway or care about alert-manager configuration.
+They simply provide an e-mail address, and they will be informed about capacity issues.
+
+A further advantage of sending alerts as e-mails through mailgun is, that we have a clear paper trail that the alert was sent and received by the service user.
```

Comment:
Technically we can't really prove an alert was received. It might be discarded by the service users mail provider.

Reply:
I'm no e-mail expert, but can't we prove that their mail sever did not bounce the e-mail and at that point it's their fault if the e-mail didn't end up at the user?

Reply:
We can proof that their server received a mail, what happens afterward is not in our power. Not sure if that's enough if this would become a legal matter...